src/components/App.tsx

```diff
@@ -422,19 +417,19 @@ export class App extends React.Component<AppProps, AppState> {
     const gistURI = await Service.exportProjectToGist(this.project, this.state.fiddle);
     this.logLn("Project Gist CREATED ");
     if (gistURI) {
-      this.setState({
-        toast:  gistURI
-      });
+      this.setState((prevState: any) => ({
+        toasts:  [...prevState.toasts, gistURI]
+      }));
       console.log(`Gist created: ${gistURI}`);
     } else {
       console.log("Failed!");
     }
   }
 
   onToastDismiss(index: number) {
-      this.setState({
-          toast:  null
-      });
+    this.setState((prevState: any) => ({
+      toasts: prevState.toasts.filter((key: number, value: number) => value !== index )
```

Comment:
I think the `Toast` component should have the toasts in its state, or in redux store(do we have it?)

Reply:
1) Action/Event is in App(Parent), so the state is kept in App.
2) No redux.

Reply:
Yeah, what I was saying is, it should be independent since we might have many components who would like to generate toasts.

Reply:
I think it's okay to have the Toast be a controlled component. We can write a ToastContainer component that wraps it and provides an immediate mode API, like `showToast(...)`; The App component can hold a reference to the ToastContainer. What do you think?